### DISCUSSION TITLE
Can a solid and/or pipeline depend on its previous run?

### DISCUSSION THREAD

**Discussion by JonathanGrant** [dagster-io/dagster#5346] [Q&A]
I want to be able to specify not to run a solid/pipeline until the previous day's run completes successfully. Is this possible?

**Comment by sryza** [comment]
Hi @JonathanGrant - it should be possible, but not trivial, to implement this with a sensor (which can contain arbitrary logic for when to run jobs).

This snippet gives a taste of how, inside a sensor, you can query Dagster for completed pipeline runs: https://github.com/dagster-io/dagster/blob/0.12.9/examples/hacker_news/hacker_news/sensors/download_pipeline_finished_sensor.py.
